backend/lib/models/Notification.js

```diff
@@ -0,0 +1,47 @@
+const { Schema, model } = require("mongoose");
+
+const notificationSchema = new Schema(
+  {
+    action: {
+      enum: ["like", "comment", "post"],
+      required: true,
+      type: String,
+    },
+    post: {
+      id: String,
+      title: String,
+    },
+    receiver: {
+      ref: "User",
+      required: true,
+      type: Schema.Types.ObjectId,
+    },
+    seenAt: Date,
+    sentAt: Date,
```

Comment:
Maybe a boolean is enough?
For seenAt at least.
Maybe we can keep sentAt as a date if we want to send the same notification on daily + weekly digest etc.. and not only once.

Reply:
Yeah, I was thinking maybe `seenAt` could be useful for BI/analytics to see interaction of users (future-proofing) but if that's definitely not something that would happen, then we can change it to a boolean and rename it to `isRead`.

Reply:
> Yeah, I was thinking maybe seenAt could be useful for BI/analytics to see interaction of users (future-proofing) but if that's definitely not something that would happen, then we can change it to a boolean and rename it to isRead.

I think we should keep it as a date for future proofing. It feels like it should be `readAt` however (maybe because `sentAt` looks too similar). This would be when the user reads clicks on the notification (or marks all as read) in the web app?

The `sentAt` is then when the lambda sends the email for that same notification correct?

Reply:
> The sentAt is then when the lambda sends the email for that same notification correct?

Yes. Maybe `sentAt` should be renamed to `emailSentAt`?

I will rename `seenAt` to `readAt`. And yes, this timestamp would be for when the notification is marked as read in the web app (and in the future, in the mobile native app).

Reply:
Sounds good! @mannykary

Reply:
I am trying to follow the idea, so who's the responsibility to update the readAt and emailSentAt? I guess FE will call BE to update the readAt once the user read the notification, will redis update the emailSentAt? (sorry if that's a silly question)

Reply:
@ddveloper The Lambda will need to filter out any notifications for which it has already sent an email, and so it can set the `emailSentAt` field after successfully sending an email for the notification. So the next time in queries for notifications, it can filter out any that have `emailSentAt` set to a value. It will also filter out any notifications for which the `readAt` field is set as well.